PostFreeSurfer/PostFreeSurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,420 @@
+opts_AddMandatory '--sessions' 'Timepoints_string' 'FS timepoint ID(s)' "Freesurfer timepoint ID(s). For timepoint (session)\
+                    processing, specify current timepoint. For template processing, must specify all timepoints, @ separated.\
+                    Timepoint ID and Session are synonyms in HCP structural pipelines."
+opts_AddMandatory '--template_processing' 'TemplateProcessing' 'create template flag' "0 if TP processing; 1 if template processing (must be run after all TP's)"
+
+#the following options have the same meaning as in PreFreesurferPipeline.
+opts_AddMandatory '--t1template' 'T1wTemplate' 'file_path' "MNI T1w template"
+opts_AddMandatory '--t1templatebrain' 'T1wTemplateBrain' 'file_path' "Brain extracted MNI T1wTemplate"
+opts_AddMandatory '--t1template2mm' 'T1wTemplate2mm' 'file_path' "MNI 2mm T1wTemplate"
+opts_AddMandatory '--t2template' 'T2wTemplate' 'file_path' "MNI T2w template"
+opts_AddMandatory '--t2templatebrain' 'T2wTemplateBrain' 'file_path' "Brain extracted MNI T2wTemplate"
+opts_AddMandatory '--t2template2mm' 'T2wTemplate2mm' 'file_path' "MNI 2mm T2wTemplate"
+opts_AddMandatory '--templatemask' 'TemplateMask' 'file_path' "Brain mask MNI Template"
+opts_AddMandatory '--template2mmmask' 'Template2mmMask' 'file_path' "Brain mask MNI 2mm Template"
+opts_AddMandatory '--fnirtconfig' 'FNIRTConfig' 'file_path' "FNIRT 2mm T1w Configuration file"
+
+#needed to transform Freesurfer mask to MNI space.
+opts_AddMandatory '--freesurferlabels' 'FreeSurferLabels' 'file' "location of FreeSurferAllLut.txt"
+
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var FSLDIR
+log_Check_Env_Var HCPPIPEDIR_Global
+
+# Naming Conventions
+
+T1wImageBrainMask="brainmask_fs"
+MNI_hires_template="$HCPPIPEDIR/global/templates/MNI152_T1_0.8mm.nii.gz"
+TemplateProcessing=$(opts_StringToBool "$TemplateProcessing")
+
+
+if [[ "$Timepoints_string" =~ "@" ]]; then 
+    if (( TemplateProcessing )); then    
+            log_Err_Abort "More than one timepoint is specified in template mode, please check calling script."
+    fi
+    IFS='@' read -r -a timepoints <<< "$Timepoints_string"
+    Timepoint_cross=${timepoints[0]}
+else
+    if (( ! TemplateProcessing )); then
+        log_Err_Abort "At least two timepoints must be specified in timepoint processing mode, please check calling script."
+    fi
+    Timepoint_cross=$timepoints
+fi
+
+#########################################################################################
+# Organizing and cleaning up the folder structure
+#########################################################################################
+if (( ! TemplateProcessing )); then 
```

Comment:
You have a lot of consecutive "not template processing" blocks with no `else` code, and the lines in between them just set some variables, at least some of which shouldn't really get set in non-template mode.  Please consider whether combining them all into a large single "not template mode" conditional would improve the script structure.

Reply:
These blocks are organized logically by operation type and order in PreFreeSurferPipeline, and merging if-clauses for some of them would break this structure. I added appropriate comment in the end of each block where no template processing is done to remind of that. 
I also moved some variable declarations only used inside such if statements to better match scope.
I also found an issue where Use_T2w is always set to 1 in template mode, fixed that.